cmd/ipfs-cluster-service/daemon.go

```diff
@@ -271,3 +273,47 @@ func setupPinTracker(
 		return nil
 	}
 }
+
+func setupDatastore(
+	consensus string,
+	cfgs *cfgs,
+) ds.ThreadSafeDatastore {
+	stmgr := newStateManager(consensus, cfgs)
+	store, err := stmgr.GetStore()
+	checkErr("creating datastore", err)
+	return store
+}
+
+func setupConsensus(
+	name string,
+	h host.Host,
+	pubsub *pubsub.PubSub,
+	cfgs *cfgs,
+	store ds.ThreadSafeDatastore,
```

Comment:
`ThreadSafeDatastore` is removed
https://github.com/ipfs/go-datastore/commit/9f529bc350a59b53118f6dcd37c7fa1be861d923